internal/keyvalue/leveldb.go

```diff
@@ -26,23 +26,38 @@ import (
 	"time"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/database"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/metrics"
 	"github.com/syndtr/goleveldb/leveldb"
 	"github.com/syndtr/goleveldb/leveldb/errors"
 	"github.com/syndtr/goleveldb/leveldb/filter"
-	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"github.com/syndtr/goleveldb/leveldb/opt"
 	"github.com/syndtr/goleveldb/leveldb/util"
 )
 
 const (
-	writePauseWarningThrottler = 1 * time.Minute
-)
+	// leveldbDegradationWarnInterval specifies how often warning should be printed
+	// if the leveldb database cannot keep up with requested writes.
+	leveldbDegradationWarnInterval = time.Minute
+
+	// leveldbMinCache is the minimum amount of memory in megabytes to allocate to
+	// leveldb read and write caching, split half and half.
+	leveldbMinCache = 16
 
-var OpenFileLimit = 64
+	// leveldbMinHandles is the minimum number of files handles to allocate to the
+	// open database files.
+	leveldbMinHandles = 16
+
+	// metricsGatheringInterval specifies the interval to retrieve leveldb database
+	// compaction, io and pause stats to report to the user.
+	metricsGatheringInterval = 3 * time.Second
+)
 
-type LDBDatabase struct {
+// LeveldbDatabase is a persistent key-value store. Apart from basic data storage
+// functionality it also supports batch writes and iterating over the keyspace in
+// binary-alphabetical order.
+type LeveldbDatabase struct {
```

Comment:
Nit: This is true across the codebase but figured I would raise it here first. Idiomatic golang capitalizes abbreviation names. So the "db" in `levelDb*` and `Leveldb*` should be `levelDB*` and `LevelDB*`. This is true with `chainDb` and `stateDB` variable names as well as names that contain subprotocols in it `Les*` should be  `LES*`.

https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
Fair enough, fixed.